src/governance.ts

```diff
@@ -133,6 +147,51 @@ export class GovernanceCanister {
     return this.responseConverters.toListProposalsResponse(rawResponse);
   };
 
+  public stakeNeuron = async ({
+    stake,
+    principal,
+    ledgerCanister,
+  }: {
+    stake: ICP;
+    principal: Principal;
+    ledgerCanister: LedgerCanister;
+  }): Promise<NeuronId | TransferError> => {
+    if (stake.toE8s() < E8S_PER_ICP) {
+      return new InsufficientAmount(ICP.fromString("1") as ICP);
+    }
+
+    try {
+      const nonceBytes = new Uint8Array(randomBytes(8));
+      const nonce = uint8ArrayToBigInt(nonceBytes);
+      const toSubAccount = this.buildSubAccount(nonceBytes, principal);
+      const accountIdentifier = AccountIdentifier.fromPrincipal({
+        principal: this.canisterId,
+        subAccount: toSubAccount,
+      });
+
+      // Send amount to the ledger.
+      const response = await ledgerCanister.transfer({
+        memo: nonce,
+        amount: stake,
+        to: accountIdentifier,
+        // TODO: Support subaccounts
+      });
+
+      if (typeof response !== "bigint") {
```

Comment:
nit: It'd be more clear if we explicitly check that the type is a `TransferError`.

Reply:
How do I do that? `TransferError` is a typescript type, not a JS class.

That's why I asked about converting `TransferError` to a class and extend all the Errors with it.